_posts/2022-01-19-bazel-5.0.md

```diff
@@ -0,0 +1,148 @@
+* Added `--bes_header` flag to pass extra headers to the BES server.
+* BEP now includes test suite expansions.
+* BEP now includes all files from successful actions in requested output groups. Previously, an output group's files were excluded if any file in the output group was not produced due to a failing action. Users can expect BEP output to be larger for failed builds.
+* In BEP, `TargetComplete.output_group` has a new field `incomplete` indicating that the `file_sets` field is missing one or more declared artifacts whose generating actions failed.
+* When `--experimental_bep_target_summary` is enabled, BEP contains a new event, `TargetSummary`. For each top-level configured target, this event aggregates the results for building the configured target, tests run for that configured target (if any), and building all top-level aspects applied to that configured target. This flag will eventually default to enabled.
+
+## C++ / Objective-C
+
+* The Starlark method `generate_dsym` in `objc` fragment has been deleted.  Please use the equivalent `apple_generate_dsym` in `cpp` fragment instead.
+* Removed `--incompatible_objc_compile_info_migration`. ([#10854](https://github.com/bazelbuild/bazel/issues/10854))
+* Removed `--incompatible_objc_provider_remove_compile_info`. ([#11359](https://github.com/bazelbuild/bazel/issues/11359))
+* Made gcov optional in `cc_toolchain` tools.
+* Added a new flag, `--incompatible_enable_cc_test_feature` which switches from the use of build variables to the feature of the same name.
+
+## Configurability
+
+* Added `--incompatible_enforce_config_setting_visibility` to make `config_setting` honor the `visibility` attribute (defaulting to `//visibility:public`) ([#12932](https://github.com/bazelbuild/bazel/issues/12932))
+* The flag `--toolchain_resolution_debug` now takes a regex argument, which is used to check which toolchain types or targets should have debug info printed. You may use `.*` as an argument to keep the current behavior of debugging every toolchain type / target.
+
+## Coverage
+
+* All (instead of just C++) source files are now filtered for coverage output according to `--instrumentation_filter` and `--instrument_test_targets`.
+* `genrule.srcs` is now considered a source attribute for coverage.
+* Forward coverage-instrumented files from non-tool dependencies by default.
+* `label_keyed_string_dict` attributes are now considered when gathering instrumented files for coverage.
+* Java branch coverage now applies Jacoco’s coverage filters. ([#12696](https://github.com/bazelbuild/bazel/issues/12696))
+
+## Java
+
+* Removed `javac_jar` from JavaToolchainInfo.
+* Flipped `--incompatible_disallow_resource_jars`, disabling `resouce_jars` attribute on `java_library`. ([#13221](https://github.com/bazelbuild/bazel/issues/13221))
```

Comment:
Rather than "flipped", which does not specify the final state, how about
'--incompatible_disallow_resource_jars is now true by default' ...
Or follow the pattern of the others and say flipped to X

Reply:
Done. ("flipped to `true`")